classes/setup_item.php

```diff
@@ -180,6 +208,18 @@ public function __construct(string $data) {
         if (!empty($this->reposition) && !empty($this->add)) {
             throw new invalid_setup_item_exception('conflictreposition');
         }
+
+        if (isset($values[$showinsubcontextindex]) && is_numeric($values[$showinsubcontextindex])) {
+            $this->showinsubcontexts = (int) trim($values[$showinsubcontextindex]);
+        }
+
+        if (isset($values[$showinsubcontextindex + 1])) {
```

Comment:
Just in case the elements get shuffled in the future, I think you should also store the pagetypepatternindex in a variable, too.

Reply:
Do you have an update on this suggestion?

Reply:
Good spot, thanks.